packages/server/src/adapters/fetch/types.ts

```diff
@@ -25,20 +25,20 @@ export type FetchCreateContextFn<TRouter extends AnyRouter> = (
   opts: FetchCreateContextFnOptions,
 ) => inferRouterContext<TRouter> | Promise<inferRouterContext<TRouter>>;
 
-export type FetchCreateContextOption<TRouter extends AnyRouter> =
-  unknown extends inferRouterContext<TRouter>
-    ? {
-        /**
-         * @link https://trpc.io/docs/v11/context
-         **/
-        createContext?: FetchCreateContextFn<TRouter>;
-      }
-    : {
-        /**
-         * @link https://trpc.io/docs/v11/context
-         **/
-        createContext: FetchCreateContextFn<TRouter>;
-      };
+export type FetchCreateContextOption<TRouter extends AnyRouter> = {
+  /**
+   * @link https://trpc.io/docs/v11/context
+   **/
+  createContext: object extends inferRouterContext<TRouter>
```

Comment:
We can use the callback helper here I think

Reply:
Ah you're right I forgot to tidy this up, I'll sort the types out

Reply:
I had just switched back this one case because it was the only one failing, but then did the internal/public types separately and missed this one